expression/helper.go

```diff
@@ -133,15 +133,20 @@ func getSystemTimestamp(ctx sessionctx.Context) (time.Time, error) {
 		return now, err
 	}
 
-	if timestampStr == "" {
-		return now, nil
-	}
-	timestamp, err := types.StrToInt(sessionVars.StmtCtx, timestampStr)
-	if err != nil {
-		return time.Time{}, err
+	if timestampStr != "" {
+		timestamp, err := types.StrToInt(sessionVars.StmtCtx, timestampStr)
+		if err != nil {
+			return time.Time{}, err
+		}
+		if timestamp <= 0 {
+			return now, nil
+		}
+		return time.Unix(timestamp, 0), nil
 	}
-	if timestamp <= 0 {
-		return now, nil
+	var nowTs = &ctx.GetSessionVars().StmtCtx.NowTs
+	//update nowTs if not set yet
+	if nowTs.Equal(time.Time{}) {
+		*nowTs = now
```

Comment:
It's better to write in a straightforward way. At first, I thought that all the places that use `NowTs` are read.